From 17ab9ed7629a8b4c89b5c3574dc12a1e644e2faf Mon Sep 17 00:00:00 2001
From: Yael Samet <yael.samet@intel.com>
Date: Wed, 23 Nov 2016 10:07:59 +0200
Subject: [PATCH 4644/4706] mei: dal: add error code for invalid ta acp file

commit b6649eefd8bce9b21f51b573181745702c60912e from
git://git.yoctoproject.org/linux-yocto-4.1

Thhis error is produced when trying to open session
with an invalid acp file.

Change-Id: I1e58a7a93c1f38fa8f220428d30822cf217ebb38
Signed-off-by: Yael Samet <yael.samet@intel.com>
Signed-off-by: Tomas Winkler <tomas.winkler@intel.com>
---
 drivers/misc/mei/dal/dal_kdi.c |    2 ++
 include/linux/dal.h            |    1 +
 2 files changed, 3 insertions(+), 0 deletions(-)

diff --git a/drivers/misc/mei/dal/dal_kdi.c b/drivers/misc/mei/dal/dal_kdi.c
index a83b075..af2faa64 100644
--- a/drivers/misc/mei/dal/dal_kdi.c
+++ b/drivers/misc/mei/dal/dal_kdi.c
@@ -109,6 +109,8 @@ static int bh_err_to_kdi_err(int bh_err)
 		return DAL_KDI_STATUS_WD_TIMEOUT;
 	case BHE_APPLET_CRASHED:
 		return DAL_KDI_STATUS_APPLET_CRASHED;
+	case BHE_TA_PACKAGE_HASH_VERIFY_FAIL:
+		return DAL_KDI_STATUS_INVALID_ACP;
 	default:
 		return DAL_KDI_STATUS_INTERNAL_ERROR;
 	}
diff --git a/include/linux/dal.h b/include/linux/dal.h
index 0e60118..e34a9f2 100644
--- a/include/linux/dal.h
+++ b/include/linux/dal.h
@@ -96,6 +96,7 @@ struct dal_version_info {
 #define DAL_KDI_STATUS_NON_EXCLUSIVENESS_TA     0xA0F
 #define DAL_KDI_STATUS_TA_EXIST                 0xA10
 #define DAL_KDI_STATUS_OPERATION_NOT_PERMITTED  0xA11
+#define DAL_KDI_STATUS_INVALID_ACP              0xA12
 
 #define DAL_KDI_INVALID_HANDLE    0
 
-- 
1.7.5.4

